httpbakery/form/form.go

```diff
@@ -123,9 +140,21 @@ func (v webPageVisitor) visitWebPage(u *url.URL) error {
 	req.Header.Set("Accept", "application/json")
 	var lm loginMethods
 	if err := v.client.Do(req, nil, &lm); err != nil {
```

Comment:
Thought for the future: if we have several kinds of supported non-interactive logins, it's somewhat wasteful to do a network round trip for each fallback when they'll all be wanting to fetch the same login methods page. We should think about ways we can avoid that.